src/stt.py

```diff
@@ -102,10 +109,22 @@ def _recognize_speech_from_mic(self):
         """
         @utils.time_it
         def whisper_transcribe(audio):
-            segments, info = self.transcribe_model.transcribe(audio, task=self.task, language=self.language, beam_size=5, vad_filter=True)
-            result_text = ' '.join(segment.text for segment in segments)
+            # if using faster_whisper (default) return based on faster_whisper's code, if not assume player wants to use server mode and send query to whisper_url set by player.
+            if self.whisper_type == 'faster_whisper':
+                segments, info = self.transcribe_model.transcribe(audio, task=self.task, language=self.language, beam_size=5, vad_filter=True)
+                result_text = ' '.join(segment.text for segment in segments)
 
-            return result_text
+                return result_text
+            # this code queries the whispercpp server set by the user to obtain the response
+            else:
+                url = self.whisper_url
+                headers = {"Authorization": f"Bearer {openai.api_key}",}
```

Comment:
only problem I have with this one is that openai API key will fly in cleartext http around your network but...this is an option for 'advanced users' and will be niche so maybe it's fine?

Reply:
could do something like
if 'api.openai' in url: 
   headers = {"Authorization": f"Bearer {openai.api_key}",}

Reply:
Good point, fixed.